src/server/services/__tests__/config.service.test.ts

```diff
@@ -0,0 +1,50 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { strictEqual } from 'assert';
+import { safeString } from 'testData';
+import { ConfigService } from '../config.service';
+
+describe('Configuration Service', function () {
+  it('reports if the app is in production', async function () {
+    process.env.NODE_ENV = 'production';
```

Comment:
>Potential issue here: Our tests normally run in NODE_ENV === 'testing', which helps for things like not sending email or connecting to different databases. Changing the NODE_ENV could then affect any tests that come after this.


That's a good point, I'd forgotten about that. 


> I think the best way to resolve this is to rewrite the config service constructor so it reads the environment from a passed object, instead of directly from process. That will also make it easier if we need to test other values in the config service in the future.

Yeah, I wonder if we should tweak it a little so that it basically just parses the `.env` file straight in there and we can just read and map arbitrary values into it rather than having a defined schema for `DB_HOST` etc. What are your thoughts?

Reply:
Just to recap our offline conversation for the official record: The `.env` file is being parsed by docker-compose, not the application code. That way we can automate building the image on one system (i.e. Travis) and let the deployment details be handled by the production system (i.e. Fargate). So the easiest way for us to pass arbitrary values into the `ConfigService` is for the constructor to accept an object, which in production will just be the value `process.env` and in testing could be any set of values we want to test against.

Reply:
Addressed in 2ce2cd5dfd7efdcc7edc407452cec9356cd90058